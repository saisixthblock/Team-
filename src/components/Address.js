import React, { Component } from "react";
import { Card,CardBody, Col, FormGroup, Input, Label,Row } from "reactstrap";

class Address extends Component{
    render(){
        return(
            <div>
                <Row>
                    <Col md={6}>
                        <FormGroup>
                            <Label htmlFor="address">Address:</Label>
                            <Input type="textarea"
                                name="address"
                                value={this.props.values.address}
                                placeholder="Enter the address"
                                onChange={(e)=>this.props.handleChange(e.target.name, e.target.value)}
                                onBlur={(e)=>this.props.handleBlur(e.target.name,e.target.value)}
                            />
                            {this.props.values.addressError ? <p style={{color:'red'}}>{this.props.values.addressError}</p> : null}
                        </FormGroup>
                    </Col>
                    <Col md={6}>
                        <FormGroup>
                            <Label htmlFor="city">City:</Label>
                            <Input type="text"
                                name="city"
                                value={this.props.values.city}
                                placeholder="Enter the city"
                                onChange={(e)=>this.props.handleChange(e.target.name, e.target.value)}
                                onBlur={(e)=>this.props.handleBlur(e.target.name,e.target.value)}
                            />
                            {this.props.values.cityError ? <p style={{color:'red'}}>{this.props.values.cityError}</p>: null}
                        </FormGroup>
                    </Col>
                </Row>
                <Row>
                    <Col>
                        <FormGroup>
                            <Label htmlFor="pincode">pincode:</Label>
                            <Input type="text"
                                name="pincode"
                                placeholder="Enter the pincode"
                                value={this.props.values.pincode}
                                onChange={(e)=>this.props.handleChange(e.target.name, e.target.value)}
                                onBlur={(e)=>this.props.handleBlur(e.target.name,e.target.value)}
                            />
                            {this.props.values.pincodeError ? <p style={{color:'red'}}>{this.props.values.pincodeError}</p> : null}
                        </FormGroup>
                    </Col>
                    <Col>
                    <FormGroup>
                        <Label htmlFor="state">State:</Label>
                        <Input type="text"
                            name="state"
                            placeholder="Enter the state"
                            value={this.props.values.state}
                            onChange={(e)=>this.props.handleChange(e.target.name, e.target.value)}
                            onBlur={(e)=>this.props.handleBlur(e.target.name,e.target.value)}
                        />
                        {this.props.values.stateError ? <p style={{color:'red'}}>{this.props.values.stateError}</p> : null}
                    </FormGroup>
                    </Col>
                </Row>    
            </div>
            
        )
    }
}
export default Address