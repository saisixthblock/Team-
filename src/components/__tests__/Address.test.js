import React from 'react'
import { shallow } from 'enzyme'
import { shallowToJson } from 'enzyme-to-json';
import Enzyme from 'enzyme'
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import Address from '../Address';
import { FormGroup, Input } from 'reactstrap';

Enzyme.configure({adapter:new Adapter() });

let sampleData={
    fullName:'abcd',
    email:'abcd@gmail.com',
    password:'abcd',
    phoneNumber:'998877',
    address:'',
    city:'hyd',
    pincode:'502032',
    state:'',
    location:'',
    gender:"",
    checkbox:'',
}

describe('Testing the Address component',()=>{
    let wrapper=shallow(<Address values={sampleData} handleChange={jest.fn()} handleBlur={jest.fn()}/>);
    it('Create the snapShot for the Address component',()=>{
        expect(shallowToJson(wrapper)).toMatchSnapshot();
    });

    it('Checking the value of the input field',()=>{
        expect(wrapper.find(Input).at(0).prop('value')).toEqual(''); // checking the value of address input field
        expect(wrapper.find(Input).at(1).prop('value')).toEqual('hyd'); // checking the value of city input field
        expect(wrapper.find(Input).at(2).prop('value')).toEqual('502032'); // checking the value of pincode input field
        expect(wrapper.find(Input).at(3).prop('value')).toEqual(''); // checking the value of state input field
    });
    it('Checking the onChange for the Address component',()=>{
        wrapper.find(Input).at(0).prop('onChange')({target : {value : 'Mac society'}});
        wrapper.find(Input).at(1).prop('onChange')({target : {value : 'hyderabad'}});
        wrapper.find(Input).at(2).prop('onChange')({target : {value : '500032'}});
        wrapper.find(Input).at(3).prop('onChange')({target : {value : 'Telangana'}});
    });
    it('Checking the onBlur of the Address component',()=>{
        wrapper.find(Input).at(0).prop('onBlur')({target : {value : 'Mac society'}});
        wrapper.find(Input).at(1).prop('onBlur')({target : {value : 'hyderabad'}});
        wrapper.find(Input).at(2).prop('onBlur')({target : {value : '500032'}});
        wrapper.find(Input).at(3).prop('onBlur')({target : {value : 'Telangana'}});
    });
    it('Checking the number of times Inpu tag is used',()=>{
        expect(wrapper.find(Input)).toHaveLength(4);
    })
    it('setting the props for Address component',()=>{
        wrapper.setProps({
            ...sampleData,
            address:'Mac Society',
            state:'telangana'
        });
    });
    // it('Checking the p tag data',()=>{
    //     expect(wrapper.find('p').exists()).toBe(true);
    // })

});