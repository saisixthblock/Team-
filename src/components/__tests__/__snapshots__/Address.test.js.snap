// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing the Address component Create the snapShot for the Address component 1`] = `
<div>
  <Row
    tag="div"
    widths={
      Array [
        "xs",
        "sm",
        "md",
        "lg",
        "xl",
        "xxl",
      ]
    }
  >
    <Col
      md={6}
      tag="div"
      widths={
        Array [
          "xs",
          "sm",
          "md",
          "lg",
          "xl",
          "xxl",
        ]
      }
    >
      <FormGroup
        tag="div"
      >
        <Label
          htmlFor="address"
          tag="label"
          widths={
            Array [
              "xs",
              "sm",
              "md",
              "lg",
              "xl",
              "xxl",
            ]
          }
        >
          Address:
        </Label>
        <Input
          name="address"
          onBlur={[Function]}
          onChange={[Function]}
          placeholder="Enter the address"
          type="textarea"
          value=""
        />
      </FormGroup>
    </Col>
    <Col
      md={6}
      tag="div"
      widths={
        Array [
          "xs",
          "sm",
          "md",
          "lg",
          "xl",
          "xxl",
        ]
      }
    >
      <FormGroup
        tag="div"
      >
        <Label
          htmlFor="city"
          tag="label"
          widths={
            Array [
              "xs",
              "sm",
              "md",
              "lg",
              "xl",
              "xxl",
            ]
          }
        >
          City:
        </Label>
        <Input
          name="city"
          onBlur={[Function]}
          onChange={[Function]}
          placeholder="Enter the city"
          type="text"
          value="hyd"
        />
      </FormGroup>
    </Col>
  </Row>
  <Row
    tag="div"
    widths={
      Array [
        "xs",
        "sm",
        "md",
        "lg",
        "xl",
        "xxl",
      ]
    }
  >
    <Col
      tag="div"
      widths={
        Array [
          "xs",
          "sm",
          "md",
          "lg",
          "xl",
          "xxl",
        ]
      }
    >
      <FormGroup
        tag="div"
      >
        <Label
          htmlFor="pincode"
          tag="label"
          widths={
            Array [
              "xs",
              "sm",
              "md",
              "lg",
              "xl",
              "xxl",
            ]
          }
        >
          pincode:
        </Label>
        <Input
          name="pincode"
          onBlur={[Function]}
          onChange={[Function]}
          placeholder="Enter the pincode"
          type="text"
          value="502032"
        />
      </FormGroup>
    </Col>
    <Col
      tag="div"
      widths={
        Array [
          "xs",
          "sm",
          "md",
          "lg",
          "xl",
          "xxl",
        ]
      }
    >
      <FormGroup
        tag="div"
      >
        <Label
          htmlFor="state"
          tag="label"
          widths={
            Array [
              "xs",
              "sm",
              "md",
              "lg",
              "xl",
              "xxl",
            ]
          }
        >
          State:
        </Label>
        <Input
          name="state"
          onBlur={[Function]}
          onChange={[Function]}
          placeholder="Enter the state"
          type="text"
          value=""
        />
      </FormGroup>
    </Col>
  </Row>
</div>
`;
